src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/Options.cs

```diff
@@ -24,26 +25,49 @@ public enum SettingFields : uint
         All = uint.MaxValue
     }
 
-    public class RequestOptions
+    public class ConfigurationSelector
     {
+        public static readonly string Any = "*";
         /// <summary>
-        /// Specific label of the key.
+        /// Keys that will be used to filter.
         /// </summary>
-        public string Label { get; set; } = null;
-
-        public ETagFilter ETag { get; set; }
-
+        /// <remarks>See the documentation for this SDK for details on the format of filter expressions</remarks>
+        public List<string> Keys { get; set; } = new List<string> { Any };
         /// <summary>
-        /// If set, then key values will be retrieved exactly as they existed at the provided time.
+        /// Labels that will be used to filter.
         /// </summary>
-        public DateTimeOffset? Revision { get; set; }
-
+        /// <remarks>See the documentation for this SDK for details on the format of filter expressions</remarks>
+        public List<string> Labels { get; set; } = new List<string> { Any };
         /// <summary>
         /// IKeyValue fields that will be retrieved.
         /// </summary>
         public SettingFields Fields { get; set; } = SettingFields.All;
-        
-        public static implicit operator RequestOptions(string label) => new RequestOptions() { Label = label };
+        /// <summary>
+        /// If set, then key values will be retrieved exactly as they existed at the provided time.
+        /// </summary>
+        public DateTimeOffset? AcceptDateTime { get; set; }
+
+        public ConfigurationSelector() { }
+
+        public ConfigurationSelector(string key, string label = default)
```

Comment:
Should we add unit-tests for all these data objects? I understand that they don't do much but for example, this constructor and Clone methods may be worth testing.

Reply:
Yeah, I'll add test for it